packages/@react-spectrum/s2/src/SelectBoxGroup.tsx

```diff
@@ -0,0 +1,300 @@
+        const childValue = childElement.props?.value || String(index);
+        
+        if (isDisabled || childElement.props?.isDisabled) {
+          disabled.add(childValue);
+        }
+      }
+    });
+    
+    return disabled.size > 0 ? disabled : undefined;
+  }, [isDisabled, childrenArray]);
+  
+  useEffect(() => {
+    if (childrenArray.length > 9) {
+      console.error('Invalid content. SelectBoxGroup cannot have more than 9 children.');
+    }
+  }, [childrenArray.length]);
+
+  const selectBoxContextValue = useMemo(
+    () => {
+      const contextValue = {
+        allowMultiSelect: selectionMode === 'multiple',
+        orientation,
+        isDisabled,
+        isCheckboxSelection,
+        selectedKeys,
+        onSelectionChange: setSelectedKeys
+      };
+      
+      return contextValue;
+    },
+    [selectionMode, orientation, isDisabled, isCheckboxSelection, selectedKeys, setSelectedKeys]
+  );
+
+  return (
+    <div
+      className={containerStyles(null, props.styles)}
+      style={UNSAFE_style}
+      ref={domRef}>
+      
+      <FormIntegration
+        name={name}
+        selectedKeys={selectedKeys}
+        selectionMode={selectionMode} />
+      
+      <ListBox
+        layout="grid"
+        selectionMode={selectionMode}
+        selectedKeys={selectedKeys}
+        onSelectionChange={setSelectedKeys}
+        disabledKeys={disabledKeys}
+        aria-label={ariaLabel || 'Select options'}
+        aria-labelledby={ariaLabelledby}
+        className={gridStyles({gutterWidth, orientation}, props.styles)}
+        style={{
+          gridTemplateColumns: `repeat(${numColumns}, 1fr)`
+        }}>
+        
+        {childrenArray.map((child, index) => {
+          if (!React.isValidElement(child)) {return null;}
+          
+          const childElement = child as ReactElement<{value?: string}>;
+          const childValue = childElement.props?.value || String(index);
+          
+          const getTextValue = (element: ReactElement): string => {
+            const elementProps = (element as any).props;
+            const children = React.Children.toArray(elementProps.children) as ReactElement[];
+            const textSlot = children.find((child: any) => 
+              React.isValidElement(child) && (child as any).props?.slot === 'text'
+            );
+            
+            if (React.isValidElement(textSlot)) {
+              return String((textSlot as any).props.children || '');
+            }
+            
+            const textContent = children
+              .filter((child: any) => typeof child === 'string')
+              .join(' ');
+            
+            return textContent || childValue;
+          };
+          
+          const textValue = getTextValue(childElement);
+          
+          return (
+            <ListBoxItem key={childValue} id={childValue} textValue={textValue} aria-label={textValue}>
```

Comment:
<img width="199" height="189" alt="Screenshot 2025-08-01 at 5 17 32 PM" src="https://github.com/user-attachments/assets/3824df7a-61f4-4f39-9331-69a50efcb0e2" />

when i keyboard focus, i get an extra outilne. might need to add another `outline: none` somewhere